pkg/trivyk8s/trivyk8s.go

```diff
@@ -277,13 +291,11 @@ func (c *client) ListSpecificArtifacts(ctx context.Context) ([]*artifacts.Artifa
 			if c.excludeOwned && c.hasOwner(resource) {
 				continue
 			}
-
 			lastAppliedResource := resource
-			if jsonManifest, ok := resource.GetAnnotations()["kubectl.kubernetes.io/last-applied-configuration"]; ok { // required for outdated-api when k8s convert resources
-				err := json.Unmarshal([]byte(jsonManifest), &lastAppliedResource)
-				if err != nil {
-					continue
-				}
+
+			if err := setActualResource(&resource, c.useActualConfig); err != nil {
+				slog.Error("Unable to get actual resource", "error", err)
+				continue
```

Comment:
Does this logic work? We store `lastAppliedResource` but mutate `resource` as it's the one passed by reference.

The tests don't test this code path.